zipline/_protocol.pyx

```diff
@@ -226,6 +229,11 @@ cdef class BarData:
             dt = \
                 self.data_portal.trading_calendar.previous_minute(dt)
 
+        if self._daily_mode:
+            # if we're in daily mode, take the given dt (which is the last
+            # minute of the session) and get the session label for it.
+            dt = self.data_portal.trading_calendar.minute_to_session_label(dt)
```

Comment:
Hard for me to know if this is the right place for this translation.  Would you mind adding some doco to this method to explain who the consumers are and what the expected return values are?